File: api/domain/entity/learner.py
from datetime import date, datetime
from typing import Optional
from pydantic.main import BaseModel
from .learner_speech import LearnerSpeech


class Learner(BaseModel):
    user_id: Optional[int]
    teacher_id: Optional[int]
    name: Optional[str]
    gender: Optional[str]
    birth_date: Optional[date]
    birth_place: Optional[str]
    year_of_learning: Optional[int]
    created_at: Optional[datetime]
    speeches: Optional[list[LearnerSpeech]]

File: ui/src/types/Learner.ts
import { LearnerSpeech } from './LearnerSpeech'

export interface Learner {
  user_id: number
  teacher_id: number
  name: string
  gender: string
  birth_date: string
  birth_place: string
  year_of_learning: number
  created_at: string
  speeches: LearnerSpeech[]
  gop_average: number
  dtw_average: number
}
